crates/engine/tree/src/tree/mod.rs

```diff
@@ -2431,11 +2433,12 @@ where
 
         // Use cached state provider before executing, used in execution after prewarming threads
         // complete
-        let state_provider = CachedStateProvider::new_with_caches(
-            state_provider,
-            handle.caches(),
-            handle.cache_metrics(),
-        );
+        let state_provider =
+            InstrumentedStateProvider::from_state_provider(CachedStateProvider::new_with_caches(
+                state_provider,
+                handle.caches(),
+                handle.cache_metrics(),
+            ));
```

Comment:
this definitely adds some overhead to all ops, this is def useful for debugging and testing but unsure if we should make this the default

but maybe the overhead is negligible